plutus-tx/compiler/Language/PlutusTx/Compiler/Expr.hs

```diff
@@ -80,6 +79,17 @@ So we use a horrible hack and match on `build . unpackFoldrCString#` to "undo" t
 rule.
 -}
 
+{- Note [Addr#]
+String literals usually have type `Addr#`. This is not very nice for us, since we certainly don't have
+those.
+
+However, they should only appear wrapped by an `unpackCString#` or similar. Except if the string literal
+gets lifted out by GHC! This is no problem for our compiler, we will just compile the literal as a binding
+and put in a reference... except that the binding would need to be of type `Addr#`.
+
+So we use another horrible hack and pretend that `Addr#` is `String`, since we treat `unpackCString#` as doing nothing.
+-}
```

Comment:
Do we want to add `Addr#` as an extensible type once we have extensible types? Or maybe a special-purpose type that somehow represents various obscure things like `Addr#` (provided we'll have more of them)?

Reply:
If we actually want to represent `Addr#`-like things, sure. But I doubt we want to do that! Most of the other `Addr#` stuff is for byte arrays and so on, it just happens that it also crops up for string literals.

It's an open question how the Plutus Tx compiler is going to deal with the target language having extensible builtins. At the moment it assumes we have `String` so it can compile string-related stuff to it, but that can just be a flag and it can give an error if it's not present.